presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -320,6 +323,15 @@ private void invokeNoMoreSplitsIfNecessary()
         return COMPLETED_FUTURE;
     }
 
+    private void handleFileHeader(Properties schema, JobConf jobConf)
+    {
+        int headerCount = getHeaderCount(schema);
+        if (headerCount > 0) {
+            // do not split file when skip.header.line.count is used
```

Comment:
Can't we skip the header and split the rest?

Reply:
We could but I would prefer not to do it in this PR, it would be much bigger change. I would need to mark split for which we would need to skip header rows.